streaming-binlog/src/main/java/org/openmrs/analytics/DebeziumListener.java

```diff
@@ -91,37 +88,32 @@ public void configure() throws Exception {
                         .log(LoggingLevel.WARN, "Event Not Supported: ${headers[" + DebeziumConstants.HEADER_IDENTIFIER + "]}")
                 .endParent();
 
-        /**
-         * FHIR event worker
-         * Current append mode is suppported i.e any Create or Update will be upserted to the warehouse
-         * append mode should also work for deletion given that openmrs does not delete data instead voids
-         * @param routeDefinition.
-         */
-        from(FHIR_ROUTE_HANDLER)
-                .filter(body().isNotNull())// Filter non-Fhir uri
+
+        // FHIR event worker
+        // Current append mode is suppported i.e any Create or Update will be upserted to the warehouse
+        // append mode should also work for deletion given that openmrs does not delete data instead voids
+        // TODO handle for manual Deletion
+        from(FHIR_HANDLER_ROUTE)
+                .routeId(DebeziumListener.class.getName() + ".FhirHandlerRoute")
+                .filter(body().isNotNull())
                 .choice()
-                    .when().simple("${header.fhirResourceUri} == null || ${header.fhirResourceUri} == ''")
+                    .when( header("fhirResourceUri").isNull())
                         .log(LoggingLevel.WARN,"FHIR URL Not Mapped --->  ${headers[" + DebeziumConstants.HEADER_IDENTIFIER + "]}")
                     .otherwise()
-                        .setBody(simple("${null}")) // Set up body to null
-                        .toD("{{openmrs.serverUrl}}{{openmrs.fhirBaseEndpoint}}${header.fhirResourceUri}" +
-                                "?httpMethod=GET" +
-                                "&authMethod=Basic" +
-                                "&authUsername={{openmrs.username}}" +
-                                "&authPassword={{openmrs.password}}" +
-                                "&authenticationPreemptive=true")
+                        .toD(getFhirConfig()).id("fhirOpenMRS") // this id is used during mocking
```

Comment:
Hi @bashir2 

Thank you for having this in-depth review and giving further direction on this. What you are proposing is very possible and can be implemented. 

I still think it is important for us to leverage on camel's API. It is good to know that if we deviate from Camel, we would be missing out much of the advantages that initially attracted us to use camel. For example, having everything inside the processor means that we will have to implement our own parallelization logic that guarantees fault tolerance for each event - which camel provides

I am trying to understand your concern, what to do you mean by  "..avoid the DSL in getFhirConfig"? Please take a look at getFhirConfig as it is just a function that generates  FHIR string URL .  Are you concerned by 
`
"&authUsername={{openmrs.username}}" +
                                "&authPassword={{openmrs.password}}" +
                                "&authenticationPreemptive=true")
`
Are you concerned by {{openmrs.username}} ? we can replace with System.getProperty("openmrs.username")

Reply:
Okay then, if we will loose benefits of Camel parallelization by moving the whole logic into `process` then please disregard this comment. I didn't think this change may have such an impact (I'll try to better understand this once this PR is merged and I play with it in my local env.).

Re. `getFhirConfig`, yes my main concern/point is relying too much on the Camel DSL, i.e., I would have preferred if we would call a common Java function that fetches a FHIR resource from OpenMRS (with all nuances like `_summary`) instead of trying to reconstruct the URL inside `getFhirConfig`. But again, following what I said above, let's leave it for now.

Reply:
I think this is a nice conversation worth following up after we are done with prototyping! And we could evaluate other strategies e.g using Beam runner + Debezium embedded engine! Though I'm not an expert on beam.

Reply:
BTW there is a camel API for parallel processing "parallelProcessing" worth looking at!

Reply:
Agreed and thanks.